news/_posts/2019-10-10-about-overriding-system.md

```diff
@@ -0,0 +1,131 @@
+
+At the time, the idea was to give a better solution to developers that used to modify directy the sources of PrestaShop to alter the right behavior for their needs.
+With overrides, the modified files were located in a specific folder, both at the Shop or Module level. This way, upgrading the shops were easier: no need to care about these modified files anymore, "PrestaShop" was able to deal with them on their own.
+
+### Please, show me an exemple!
+
+A [recent request](https://github.com/PrestaShop/PrestaShop/pull/11910) from a community member was to update some classes to allow module developers to alter both the Customer form and Customer Form formatter.
+
+We provide native (or Core) classes in PrestaShop, named *CustomerForm* and *FormFormatter* but our implementation doesn't fits his client needs. So his strategy was to override the 2 classes and use their own, extending the core class to not have to rewrite everything.
+
+We didn't discuss details that much the developers needs, but we can imagine he wants to be able to override this function in charge of defining the Customer form:
+
+```php
+<?php
+// override/classes/form/CustomerFormatter
+use CustomerFormatter as NativeCustomerFormatter;
+
+class CustomerFormatter extends NativeCustomerFormatter
+{
+    public function getFormat()
+    {
+        // Here is the business logic to create the form
+        // with the right fields.
+    }
+}
+```
+
+Doing that allow this class to be used as a replacement of the original class in the Core.
+
+But... the core implementation is not "extensible" enough: this is a bug! Most of its properties are private and a function named `addConstraints` (in charge of  the Form validation) is private!
+
+Also, he'd like to use the translator directly from the customer.
+
+His request was reasonnable, but we won't be able to accept it.
+
+### The end of the overriding system
+
+Overrides are here since 9 years, and we now have a lot of feedbacks about it. Because developers are allowed to alter behavior of Core from their modules and because we are unable to ensure that Core update won't break updates from their modules, we didn't have fixed the "upgrade terrible mistake" issue, we only have taken the responsibility of it.
+
+The truth is that the only person able to ensure that its override won't break when the core will be upgraded is the person in charge of the override. In nine years, upgrading a shop is still a terrible experience and most of the times people don't upgrade their shop and keep their business outdated (technical point of view). We have worked hard on the upgraded module, describing more and more information about what could go wrong during the upgrade: databases migrations, non existing/deprecated classes functions/cache issues and so on.
+
+> We can say that undoubtely, Overriding system was a bad idea and we
+> are deprecating it.
```

Comment:
It can be completed, we have a lot of history of broken shops, a lot of links from comments in the forums about it. We can also describe many use cases where overrides are misused (typicaly to break the behavior of a class while hoping the  "common part" can manage our specific use case).

You tell me :+1: